core/pta/bcm/wdt.c

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019, Broadcom
+ */
+
+#include <drivers/gic.h>
+#include <drivers/sp805_wdt.h>
+#include <io.h>
+#include <kernel/interrupt.h>
+#include <kernel/misc.h>
+#include <kernel/pseudo_ta.h>
+#include <trace.h>
+
+#define SEC_WD_SERVICE_UUID \
+		{ 0x6272636D, 0x2019, 0x0801,  \
+		{ 0x42, 0x43, 0x4D, 0x5F, 0x57, 0x44, 0x54, 0x30 } }
+
+#define PTA_BCM_SEC_WD_CMD_CONFIG	0
+#define PTA_BCM_SEC_WD_CMD_START	1
+#define PTA_BCM_SEC_WD_CMD_PING		2
+#define PTA_BCM_SEC_WD_CMD_STOP		3
+#define PTA_BCM_SEC_WD_CMD_SET_TIMEOUT	4
+
+#define SEC_WD_TA_NAME		"pta_bcm_sec_wd.ta"
+
+static struct sp805_wdt_data wd_pd;
+
+static void wd_isr_handler(struct wdt_chip *chip __unused)
+{
+	/* Do nothing */
+	DMSG("Watchdog ISR handler !!!");
+}
```

Comment:
I forgot to ask: what's the point of all this if the interrupt handler does nothing?

Reply:
For now, nothing to be done in the handler. I will remove this.

Reply:
Sure.
But, do you still want to merge this code now? Or would you prefer to re-submit once the implementation is complete? If it helps to have the PTA in place although it is not fully functional, then it's fine, I can merge it. I just want to clarify.

Reply:
Yes, I want to get this merged

Reply:
All right.